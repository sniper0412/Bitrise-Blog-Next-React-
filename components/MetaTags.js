import React from 'react';
import PropTypes from 'prop-types';
import Head from 'next/head';

const MetaTags = ({ title, description, ogTitle, ogDescription, pathname, image }) => (
  <Head>
    <title>{title && `${title} | `}Bitrise Blog</title>
    <meta name="description" content={description} />
    <link rel="canonical" href={`https://blog.bitrise.io/${pathname}`} />

    <meta property="fb:app_id" content="649161285439710" />
    <meta property="og:title" content={ogTitle} />
    <meta property="og:description" content={ogDescription} />
    <meta property="og:url" content={`https://blog.bitrise.io/${pathname}`} />
    <meta property="og:site_name" content="Bitrise Blog" />
    <meta property="og:type" content="article" />
    <meta property="og:image" content={image} />
    <meta property="og:image:width" content="1910" />
    <meta property="og:image:height" content="1000" />
  </Head>
);

MetaTags.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  pathname: PropTypes.string,
  image: PropTypes.string
};

MetaTags.defaultProps = {
  ogTitle: 'Bitrise Blog - Mobile Continuous Integration and Delivery',
  description:
    'Articles about Continuous Integration & Delivery, automation, iOS code signing, DevOps, and awesome tech',
  ogDescription:
    'Bitrise Blog - Mobile Continuous Integration and Delivery for your whole team, with dozens of integrations for your favourite services.',
  pathname: '',
  image: 'https://www.bitrise.io/assets/placeholders/website-social-embed.png'
};

export default MetaTags;
