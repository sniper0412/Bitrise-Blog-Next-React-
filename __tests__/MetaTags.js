import renderer from 'react-test-renderer';

import MetaTags from '../components/MetaTags';

describe('MetaTags component', () => {
  console.warn('FIXME: Snapshot does not get rendered');

  it('renders correctly with defaults', () => {
    const tree = renderer.create(<MetaTags />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom props', () => {
    const tree = renderer
      .create(
        <MetaTags
          title="Title"
          description="Simple description"
          ogTitle="Open Graph Title"
          ogDescription="Open Graph Description"
          pathname="/whatever?asd"
          image="http://meta.img"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
