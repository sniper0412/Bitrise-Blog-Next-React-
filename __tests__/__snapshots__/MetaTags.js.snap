// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaTags component renders correctly with custom props 1`] = `null`;

exports[`MetaTags component renders correctly with defaults 1`] = `null`;
